server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -336,8 +336,9 @@ public boolean shouldCache(Query query) {
         }
         indexShardOperationPermits = new IndexShardOperationPermits(shardId, threadPool);
         searcherWrapper = indexSearcherWrapper;
-        pendingPrimaryTerm = indexSettings.getIndexMetaData().primaryTerm(shardId.id());
-        operationPrimaryTerm = pendingPrimaryTerm;
+        final long primaryTerm = indexSettings.getIndexMetaData().primaryTerm(shardId.id());
+        this.pendingPrimaryTerm = primaryTerm;
+        replicationTracker.setOperationPrimaryTerm(primaryTerm);
```

Comment:
can we set this in the constructor of ReplicationTracker? Then we would always have a proper operation term in ReplicationTracker.

Reply:
I pushed 72e2e7b15e7cc3d981f956c8b015e92ba7ccf19d.